File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    
    for height in range(1, int((yellow+brown)**0.5)+1):
        if (brown+yellow) % height == 0:
            width = (brown + yellow) // height
            if (width-2) * (height-2) == yellow:
                return [width, height]

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];

        int width = brown + yellow;  
        
        //(가로-2) × (세로-2) = yellow
        for(int x=3; x<=width; x++) {
            for(int y=3; y<=width; y++) {
                if(x * y == width && (x-2) * (y-2) == yellow){
                    return new int[]{y, x};
                }
            }
        }  
        return answer;
    }
}